Source/JavaScriptCore/runtime/VM.cpp

```diff
@@ -1626,6 +1626,9 @@ void VM::visitAggregateImpl(Visitor& visitor)
     m_builtinExecutables->visitAggregate(visitor);
     m_regExpCache->visitAggregate(visitor);
 
+    if (heap.collectionScope() != CollectionScope::Full)
+        stringReplaceCache.visitAggregate(visitor);
```

Comment:
It feels a little icky that this check isn't in `StringReplaceCache::visitAggregateImpl` but we also don't have the check in the `clear` so idk.